### PR TITLE
Coalition Miners

### PR DESCRIPTION
**Pull request by Arachi-Lover**
----------------------
**Content (Artwork / Fleets)**

## Summary
This PR adds mining to Coalition space, with the primary intention of increasing visual variety to the region, with civilian ships now doing more than just travelling around.
In the Advanced Coalition outfitters, there is now a drill-esque outfit with negligible offensive potential, but good mining damage. It is used by the mining fleets to quickly break down mineables.
Said fleets are now in most Coalition systems, basically everywhere in the Coalition with mineables (I intentionally left out the two Heliarch systems with mineables in them). They are separated into three fleets, one per species, and consist of variants of all civilian Coalition ships, equipped with a pair of the mining drill outfits (except for the Saryd Traveler, which has 4 of them. Because it has 4 gun ports. For some reason.) and making use of Cogeneration Modules for some more consistent energy generation.
The fleets have both `pacifist` and `mining` as personalities, and as tested yesterday turns out the former doesn't override the latter, so they will mine about, but won't try to fight back with drills that can't even bring down the smallest human shield generator.

The mining outfit has `"asteroid scan power" 18` built-in, so that there *is* some asteroid scanning available for civilians to buy, in the drill itself. This is due to there not being another option for civilians to use to mine, so might as well have the scanning part integrated into the one "weapon" they can use for this.
This also adds `"asteroid scan power" 36` to the Saryd Traveler, translating into an innate 600 range of asteroid scan. This is due to the same reasoning that the Strong Wind was given built-in asteroid scanning, that is, the Saryd Traveler was originally a science vessel.

The outfit model was done by @Saugia recently on a livestream, and the sound effect was also provided by them. Thanks again!

## Artwork Checklist
 - [x] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [x] I created a PR to the [endless-sky-assets repo](https://github.com/EndlessSkyCommunity/endless-sky-assets) with the necessary image, blend, and texture assets: https://github.com/EndlessSkyCommunity/endless-sky-assets/pull/84
 - [x] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: https://github.com/endless-sky/endless-sky-high-dpi/pull/305

### CODE REVIEW

**Review by tibetiroka — APPROVED** [review]
Grammar looks OK.

File: data/coalition/coalition weapons.txt [hunk 1]
```diff
@@ -402,3 +402,41 @@ effect "ion rain impact"
...
+	sprite effect/smoke
+		"no repeat"
+		scale 0.5
+	sound "explosion tiny"
+	lifetime 12
+	"velocity scale" -0.05
+	"random angle" 360
+	"random velocity" 2
+	"random spin" 2
+	"random frame rate" 2
```
Comment (pilover100) [inline data/coalition/coalition weapons.txt:442]: Given that this is a civilian outfit with little to no offensive capabilities, would this be better under the generic outfits file as this isn't a weapon per se? Just a question over whether or not it should be moved but it's fine leaving it in here if wished.

File: data/coalition/coalition weapons.txt [hunk 2]
```diff
@@ -402,3 +402,41 @@ effect "ion rain impact"
...
+		"velocity" 10
+		"lifetime" 1
+		"reload" 1
+		"firing energy" .78
+		"firing heat" .68
+		"shield damage" .2
+		"hull damage" 1.4
+		"minable damage" 8.2
+	description `Not legally a weapon under Heliarch definitions, the "Mineral Collection and Sorting" extraction unit consists of a set of sharp, durable blades designed to quickly break down mineable asteroids.`
+	description `	Especially popular with novice pilots looking to pay off their ships, the extractor has been a reliable tool of the Coalition mining business for millennia.`
```
Comment (pilover100) [inline data/coalition/coalition weapons.txt:430]: This line makes it seem like there's a singular central mining guild/house in the Coalition? just questioning whether this is the case. If not a better description may be:
```suggestion
	description `	Especially popular with novice pilots looking to pay off their ships, the extractor has been a reliable tool of mining businesses in the Coalition for millennia.`
```

**Review by pilover100 — COMMENTED** [review]
Overall, it's a good addition to the coalition with the introduction of mining vessels without undermining the existing law of no civilian weapons. Skimming over it I can't spot any showstoppers though I have a couple of questions on a few points it should be fine either way.

These mining only outfits really add some more depth into the game than just the old weapons of war doing everything.

File: data/coalition/coalition weapons.txt (hunk 1 above)
Comment (Arachi-Lover) [reply data/coalition/coalition weapons.txt:442]: It's still a weapon to the game (`Gun`, to be specific), so I felt it should be with the other weapons.
Also, while it is pretty terrible at it, it **can** be used for combat, similar to how in other games we have some joke-weapons, like the Ladle in Dark Souls 2.

**Review by beccabunny — APPROVED** [review]

**Review by EjoThims — APPROVED** [review]

**Review by roadrunner56 — APPROVED** [review]

**Comment by williaji** [comment]
This PR feels like a truly Coalition slant on mining. The fact that they have a "crippled," they might say "optimized," mining gun that cannot be used as a weapon tells you much about their thought processes.

1. It took me a bit to be sure I was seeing miners in the system. This is due to me being conditioned to look for beams of light rather than puffs of regolith. This, combined with the variety of mining ships can render them invisible. If you are just flying through, you probably won't notice them at all because the visual effects are quite subtle. You will, of course, see the outfits and that might make you look for them.
2. The need to make physical contact makes for a different mining experience. Using a Kimek Thistle I found it straight forward on moderately moving silver asteroids. It was easy to lose contact but you almost never had to chase the minerals down. It felt easier with less frustration and wasted time compared to finding the range for beam convergence and then maneuvering to pick up the minerals. 
3.  I missed the lack of visual feedback when "firing" the weapon ﻿using tab. It felt very odd for nothing to appear to be happening. The audio file was there and worked well enough, but I generally play with the sound switched off. If you could add a visual indicator, I think it would be helpful. It doesn't need to be much, maybe just flashes of white  to show the drill working.
4. Gun placement matters. I tried putting 4 of the MCE extractors on a Leviathan and it was nearly impossible to mine. The asteroid scanner helps by centering you on the target but the gun ports are offset and you have to make physical contact. The end result was a large number of passes to make contact and I could never bring more than 2 extractors to bear due to their physical location. This is just and observation; I don't think there is anything to fix here. A player would need to pick the right ship and would quickly realize that more than two gun ports probably doesn't help you.
5. There is no asteroid scanner in the outfitter. Even though I doubt most folks will mine at all by this stage of the game, there should be one.

File: data/coalition/coalition weapons.txt [hunk 3]
```diff
@@ -402,3 +402,42 @@ effect "ion rain impact"
...
+		"hit effect" "drilling"
+		"velocity" 10
+		"lifetime" 1
+		"reload" 1
+		"firing energy" .78
+		"firing heat" .68
+		"shield damage" .2
+		"hull damage" 1.4
+		"minable damage" 8.2
+	description `Not legally a weapon under Heliarch definitions, the "Mineral Collection and Sorting" extraction unit consists of a set of sharp, durable blades designed to quickly break down mineable asteroids. It also comes with an integrated scanning system that looks for asteroids rich in minerals, to facilitate the mining efforts.`
```
Comment (Hecter94) [inline data/coalition/coalition weapons.txt:430]:
```suggestion
	description `Not legally a weapon under Heliarch definitions, the "Mineral Collection and Sorting" extraction unit consists of a set of sharp, durable blades designed to quickly break down minable asteroids. It also comes with an integrated scanning system that looks for asteroids rich in minerals to facilitate the mining efforts.`
```
While mineable and minable are both correct, the game consistently uses minable.

**Pull request merged**